config/backend-config/backend-config.go

```diff
@@ -277,10 +124,15 @@ func filterProcessorEnabledDestinations(config ConfigT) ConfigT {
 	return modifiedConfig
 }
 
-func configUpdate(eb *pubsub.PublishSubscriber, statConfigBackendError stats.RudderStats, workspaces string) {
-	sourceJSON, ok := backendConfig.Get(workspaces)
-	if !ok {
+func (bc *CommonBackendConfig) configUpdate(ctx context.Context, statConfigBackendError stats.RudderStats, workspaces string) {
+	sourceJSON, err := backendConfig.Get(ctx, workspaces)
+	if err != nil {
 		statConfigBackendError.Increment()
+		select {
+		case <-ctx.Done():
+		case bc.pollingErrors <- err:
```

Comment:
Good point, I'll review the logic and let you know what I can come up with :+1:

Reply:
I've added a leaky bucket approach since I don't really care if we got more errors in a row, I just want to terminate on the first one. I'll do some more tests to check things and add more coverage :+1: